storage/StorageExample/Shared/ContentView.swift

```diff
@@ -0,0 +1,261 @@
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import FirebaseAuth
+import FirebaseStorage
+
+struct ContentView: View {
+  @EnvironmentObject var vm: ViewModel
+  @State private var authenticated: Bool = true
+  private var storage = Storage.storage()
+  private var imageURL: URL = FileManager.default.temporaryDirectory
+    .appendingPathComponent("tempImage.jpeg")
+
+  var body: some View {
+    ZStack {
+      VStack {
+        if let image = vm.image {
+          image
+            .resizable()
+            .scaledToFit()
+            .frame(minWidth: 300, maxHeight: 200)
+            .cornerRadius(16)
+
+        } else {
+          Image(systemName: "photo.fill")
+            .resizable()
+            .scaledToFit()
+            .opacity(0.6)
+            .frame(width: 300, height: 200, alignment: .top)
+            .cornerRadius(16)
+            .padding(.horizontal)
+        }
+        Button("Photo") {
+          vm.showingImagePicker = true
+        }
+        .buttonStyle(OrangeButton())
+        .disabled(!authenticated)
+        if vm.image != nil {
+          Button("Upload from Data") {
+            uploadFromData()
+          }
+          .buttonStyle(OrangeButton())
+        }
+
+        if let _ = vm.image, FileManager.default.fileExists(atPath: imageURL.path) {
+          Button("Upload from URL") {
+            uploadFromALocalFile()
+          }
+          .buttonStyle(OrangeButton())
+        }
+        if vm.downloadPicButtonEnabled {
+          Button("Download") {
+            download()
+          }
+          .buttonStyle(OrangeButton())
+        }
+      }
+      .sheet(isPresented: $vm.showingImagePicker) {
+        ImagePicker(image: $vm.inputImage, imageURL: imageURL)
+      }
+      .sheet(isPresented: $vm.downloadDone) {
+        if let image = vm.downloadedImage {
+          image
+            .resizable()
+            .scaledToFit()
+            .frame(minWidth: 0, maxWidth: .infinity)
+        }
+      }
+      .onChange(of: vm.inputImage) { _ in
+        loadImage()
+      }
+      .onAppear {
+        firebaseAuth()
+      }
+      .alert("Error", isPresented: $vm.errorFound) {
+        Text(vm.errInfo.debugDescription)
+        Button("ok") {}
+      }
+      .alert("Image was uploaded", isPresented: $vm.fileUploaded) {
+        Button("ok") {}
+        Button("Link") {
+          if let url = vm.fileDownloadURL {
+            print("downloaded url: \(url)")
+            UIApplication.shared.open(url)
+          }
+        }
+      }
+
+      if vm.isLoading {
+        LoadingView()
+      }
+    }
```

Comment:
Should this method set `showingImagePicker` to false when the image picker is dismissed?